### PR TITLE
33 backend for flagging feature

### PR DESCRIPTION
**Pull request by ZeynepDYilmaz**

### CODE REVIEW

File: backend/src/main/java/com/swe573/socialhub/controller/ServiceController.java [hunk 1]
```diff
@@ -101,4 +102,25 @@ public void Complete(Principal principal, @PathVariable Long serviceId) {
             throw new ResponseStatusException(HttpStatus.BAD_REQUEST, e.getLocalizedMessage());
         }
     }
+
+    @GetMapping("/flag/{serviceId}")
```
Comment (sgbasaraner) [inline backend/src/main/java/com/swe573/socialhub/controller/ServiceController.java:106]: a resource (like flags) should be added with a POST request

File: backend/src/main/java/com/swe573/socialhub/controller/ServiceController.java [hunk 2]
```diff
@@ -101,4 +102,25 @@ public void Complete(Principal principal, @PathVariable Long serviceId) {
...
+        try {
+            Flag response = serviceService.flagService(principal, serviceId);
+            return ResponseEntity.ok(response);
+        }
+        catch (Exception e) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, e.getLocalizedMessage());
+        }
+    }
+
+    @GetMapping("/flag/control/{serviceId}")
```
Comment (sgbasaraner) [inline backend/src/main/java/com/swe573/socialhub/controller/ServiceController.java:117]: we can check if a flag exists or not via a GET request

**Review by sgbasaraner — CHANGES_REQUESTED** [review]

File: backend/src/main/java/com/swe573/socialhub/controller/ServiceController.java (hunk 2 above)
Comment (ZeynepDYilmaz) [reply backend/src/main/java/com/swe573/socialhub/controller/ServiceController.java:117]: done :) 5bb20c01d3d0c77d69ff3b54967f64e53130a6f0

File: backend/src/main/java/com/swe573/socialhub/controller/ServiceController.java (hunk 1 above)
Comment (ZeynepDYilmaz) [reply backend/src/main/java/com/swe573/socialhub/controller/ServiceController.java]: done :) 5bb20c01d3d0c77d69ff3b54967f64e53130a6f0

**Review by sgbasaraner — APPROVED** [review]

**Review by zynpgltkn — APPROVED** [review]

**Pull request merged**